WordPress/gutenberg
# [RNMobile] Fix missing title for some unsupported blocks

`gutenberg-mobile` PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3489





## Description

Block titles are now translated on the fly when they're registered so unsupported blocks, which are not registered, no longer will have the title translated.

This PR introduces the function `getTitle` into the missing block to translate the title the same way we're doing it for registered blocks.

**Code references:**
- [Translate block settings upon registration](https://github.com/WordPress/gutenberg/blob/9014efdb1ec89ad2f3460d224d279af250ed6967/packages/blocks/src/api/registration.js#L408-L419)
- [Translate using i18n schema](https://github.com/WordPress/gutenberg/blob/9014efdb1ec89ad2f3460d224d279af250ed6967/packages/blocks/src/api/registration.js#L333-L336)

## How has this been tested?




**NOTE:** Only core blocks ([defined here](https://github.com/WordPress/gutenberg/blob/feecdae6210a14d166001728dd284252bc01ffed/packages/block-library/src/index.native.js#L69-L115)) will display the title, the rest will show the block's name (i.e.: `jetpack/gif`).

1. Add some unsupported blocks to a post/page
2. Add a Table block, which is also unsupported (use the example below)
```

<figure class="wp-block-table"><table><tbody><tr><td>1</td><td>2</td></tr><tr><td>3</td><td>4</td></tr></tbody></table></figure>

```
3. Open the post/page in the mobile app
4. Observe that the table block is displaying the block's title
5. Tap on the table block
6. Observe that the bottom-sheet displays the block's title

Since this change is also related to translations, we should also perform a test using a different device's language:

1. Open device settings
2. Change to a non-English language
3. Open the post/page with unsupported blocks
4. Observe that the table block is displaying the block's title in the specified language
5. Tap on the table block
6. Observe that the bottom-sheet displays the block's title in the specified language

## Screenshots 

Unsupported blocks|Bottom-sheet opened
--|--
<img src=https://user-images.githubusercontent.com/14905380/117954939-cbd0a380-b317-11eb-9ffa-897b21e88ed1.png width=500>|<img src=https://user-images.githubusercontent.com/14905380/117954976-d7bc6580-b317-11eb-9866-6c5a0c6cd0f7.png width=500>

**In a non-English language:**
Unsupported blocks|Bottom-sheet opened
--|--
<img src=https://user-images.githubusercontent.com/14905380/117955684-94aec200-b318-11eb-8476-83db102988d9.png width=500>|<img src=https://user-images.githubusercontent.com/14905380/117955689-95dfef00-b318-11eb-9b92-2593360ff9e2.png width=500>

## Types of changes




Bug fix

## Checklist:
